vkcs/compute/data_source_flavor.go

```diff
@@ -137,100 +232,121 @@ func dataSourceComputeFlavorRead(ctx context.Context, d *schema.ResourceData, me
-					continue
-				}
+			if match {
+				filteredFlavors = append(filteredFlavors, flavor)
 			}
-
-			if v, ok := d.GetOk("swap"); ok {
-				if flavor.Swap != v.(int) {
-					continue
-				}
-			}
-
-			if v, ok := d.GetOk("rx_tx_factor"); ok {
-				if flavor.RxTxFactor != v.(float64) {
-					continue
-				}
-			}
-
-			filteredFlavors = append(filteredFlavors, flavor)
 		}
 
 		allFlavors = filteredFlavors
 	}
 
+	diags := diag.Diagnostics{}
+	if requiredFlavor.HasMinDisk && requiredFlavor.HasDisk {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Warning,
+			Summary:  "Don't set min_disk and disk together, min_disk will be overwritten by disk",
+			AttributePath: cty.Path{
+				cty.GetAttrStep{Name: "min_disk"},
+			},
+		})
+	}
+	if requiredFlavor.HasMinRAM && requiredFlavor.HasRAM {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Warning,
+			Summary:  "Don't set min_ram and ram together, min_ram will be overwritten by ram",
+			AttributePath: cty.Path{
+				cty.GetAttrStep{Name: "min_ram"},
+			},
+		})
+	}
+
 	if len(allFlavors) < 1 {
-		return diag.Errorf("Your query returned no results. " +
-			"Please change your search criteria and try again.")
+		return append(diags, diag.Errorf("Your query returned no results. "+
+			"Please change your search criteria and try again.")...)
+	}
+
+	// if we find many flavors and the user sets the min_ram or min_disk values
+	// we give him the flavor with the minimum amount of RAM from the found flavors
+	if len(allFlavors) > 1 && (requiredFlavor.HasMinRAM || requiredFlavor.HasMinDisk) {
+		resIdx := 0
+		for idx, flavor := range allFlavors {
+			if flavor.RAM == allFlavors[resIdx].RAM && flavor.Disk < allFlavors[resIdx].Disk {
+				resIdx = idx
+			} else if flavor.RAM < allFlavors[resIdx].RAM {
+				resIdx = idx
+			}
+		}
+
+		return append(diags, diag.FromErr(dataSourceComputeFlavorAttributes(d, computeClient, &allFlavors[resIdx].Flavor, allFlavors[resIdx].ExtraSpecs))...)
 	}
 
 	if len(allFlavors) > 1 {
 		log.Printf("[DEBUG] Multiple results found: %#v", allFlavors)
-		return diag.Errorf("Your query returned more than one result. " +
-			"Please try a more specific search criteria")
+		return append(diags, diag.Errorf("Found %d available flavors. Please try a more specific search criteria", len(allFlavors))...)
 	}
 
-	return diag.FromErr(dataSourceComputeFlavorAttributes(d, computeClient, &allFlavors[0]))
+	return append(diags, diag.FromErr(dataSourceComputeFlavorAttributes(d, computeClient, &allFlavors[0].Flavor, allFlavors[0].ExtraSpecs))...)
 }
 
 // dataSourceComputeFlavorAttributes populates the fields of a Flavor resource.
-func dataSourceComputeFlavorAttributes(d *schema.ResourceData, computeClient *gophercloud.ServiceClient, flavor *flavors.Flavor) error {
+func dataSourceComputeFlavorAttributes(d *schema.ResourceData, computeClient *gophercloud.ServiceClient,
+	flavor *flavors.Flavor, extraSpecs map[string]interface{}) error {
```

Comment:
You can just use argument of type of structure with ExtraSpecs field here instead of passing two arguments.